R/zzz.R

```diff
@@ -14,3 +14,4 @@
 
 # use non-exported function from teal.code
 format_expression <- getFromNamespace("format_expression", "teal.code")
+get_code_dependency <- getFromNamespace("get_code_dependency", "teal.code")
```

Comment:
I think it should be moved to this package unless you implement `get_code(names)` in `teal.code`.

Reply:
I tend to agree.

Reply:
We once had `get_code(qenv, names)` in the prototype phase of code parser. If `names` were specified then `get_code` used `get_code_dependencies`. This was meant to work on `qenv` class, that's why it was developed in `teal.code`. However, we then removed it, so that we do not change too much in the user interface. We specifically left `get_code_dependencies` non-exported, so that it could be only used in `teal.data::get_code.teal_data` in the future. If we plan to let it work only on `teal_data` which is introduced in `teal.data` I think it's fine enough to move `get_code_dependencies` from `teal.code` to `teal.data`. Will move it during this PR.

Reply:
Currently it is moved from `teal.code` to `teal.data` ae15fcd27e3c5490d992236dfe87d25ac5175f0f
I also moved an issues for `get_code_dependency` simplifications from `teal.code` to `teal.data` https://github.com/insightsengineering/teal.data/issues/192- maybe it's a time to rethink the implementation and simplify it?

Reply:
Out of scope, create a separate issue.

Reply:
hmmm, you are right, it will be cleaner if we separate

Reply:
Created an issue for that https://github.com/insightsengineering/teal.data/issues/193 and a PR https://github.com/insightsengineering/teal.data/pull/194